web/models/class/class_helper.ex

```diff
@@ -5,9 +5,32 @@ defmodule CoursePlanner.ClassHelper do
   use CoursePlanner.Web, :model
 
   alias CoursePlanner.{Repo, Class, Notifier, Attendance, Notifier.Notification}
-  alias Ecto.DateTime
+  alias CoursePlanner.Terms.Term
+  alias Ecto.{Changeset, DateTime, Date}
   alias Ecto.Multi
 
+  def validate_for_holiday(%{valid?: true} = changeset) do
+    class_date = changeset |> Changeset.get_field(:date) |> Date.cast!
+    offered_course_id = changeset |> Changeset.get_field(:offered_course_id)
+
+    term = Repo.one(from t in Term,
+      join: oc in assoc(t, :offered_courses),
+      preload: [offered_courses: oc],
```

Comment:
I am not sure I was clear, what I mean is the following
```elixir
term = Repo.one(from t in Term,
  join: oc in assoc(t, :offered_courses),
  where: oc.id == ^offered_course_id)
```
and I don't see why this would tell you that "oc" is an unbound variable